logstreams/src/main/java/io/zeebe/logstreams/impl/LogBlockIndexWriter.java

```diff
@@ -136,38 +139,30 @@ protected void onActorStarting() {
             .label("logName", getName())
             .create();
 
-    recoverBlockIndex();
-  }
-
-  private void recoverBlockIndex() {
     try {
-      final ReadableSnapshot lastSnapshot = snapshotStorage.getLastSnapshot(name);
-      if (lastSnapshot != null) {
-        lastSnapshot.recoverFromSnapshot(blockIndex);
-
-        final long snapshotPosition = lastSnapshot.getPosition();
-
-        final long snapshotBlockAddress = blockIndex.lookupBlockAddress(snapshotPosition);
-        if (snapshotBlockAddress >= logStorage.getFirstBlockAddress()) {
-          nextAddress = snapshotBlockAddress;
-          lastBlockAddress = snapshotBlockAddress;
-          lastBlockEventPosition = snapshotPosition;
-          snapshotEventPosition = snapshotPosition;
-        } else {
-          LOG.warn("Can't find address of snapshot position. Rebuild block index.");
-        }
-      }
-
-      if (nextAddress == INVALID_ADDRESS) {
-        blockIndex.reset();
-
-        nextAddress = logStorage.getFirstBlockAddress();
-        lastBlockAddress = 0;
-      }
+      blockIndex.recoverFromSnapshot(name);
     } catch (Exception e) {
       LOG.error("Failed to recover block index", e);
       throw new RuntimeException("Failed to recover block index", e);
     }
+    blockIndex.openDb();
```

Comment:
why you moved this outside the try catch ? :sweat_smile:

Reply:
Yeah, that's wrong in more than one way =p I rewrote it to catch an exception in the snapshot recovery, attempt to rebuild and then fail if an exception occurs during the rebuild